### PR TITLE
[3.3] Finishing setTrustedProxies() changes & trusted_proxies removal

### LINKED ISSUE #7671: [3.3] Changes to trusted_proxies config
See https://github.com/symfony/symfony/pull/21830

### PR DESCRIPTION
**Pull request by weaverryan**
Finishes #7868 and closes #7671

### CODE REVIEW

File: _build/redirection_map [hunk 1]
```diff
@@ -336,3 +336,4 @@
 /components/dependency_injection/autowiring /service_container/autowiring
 /event_dispatcher/class_extension /event_dispatcher
 /security/target_path /security
+/components/http_foundation/trusting_proxies /request/load_balancer_reverse_proxy
```
Comment (xabbuh) [inline _build/redirection_map]: missing trailing newline

File: http_cache/varnish.rst [hunk 2]
```diff
@@ -20,24 +20,9 @@ Varnish automatically forwards the IP as ``X-Forwarded-For`` and leaves the
...
-    sub vcl_recv {
-        unset req.http.Forwarded;
-    }
-
-If you do not have access to your Varnish configuration, you can instead
-configure Symfony to distrust the ``Forwarded`` header as detailed in
-:ref:`How to Configure Symfony to Work behind a Load Balancer or a Reverse Proxy <request-untrust-header>`.
+Remember to call the :ref:`Request::setTrustedProxies() <request-set-trusted-proxies>`
+method in your front controller so that Varnish is seen as a trusted proxy
+and the :ref:`X-Forwarded <varnish-x-forwarded-headers>` headers are used.
```
Comment (xabbuh) [inline http_cache/varnish.rst]: `X-Forward-*`?

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -183,44 +182,13 @@ named ``kernel.http_method_override``.
         $request = Request::createFromGlobals();
         // ...
 
+
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: should be reverted

File: request/load_balancer_reverse_proxy.rst [hunk 4]
```diff
@@ -7,66 +7,45 @@ an AWS Elastic Load Balancer) or a reverse proxy (e.g. Varnish for
...
 
-        <!-- app/config/config.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:framework="http://symfony.com/schema/dic/symfony"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                http://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+    // tell Symfony about your revers proxy
```
Comment (xabbuh) [inline request/load_balancer_reverse_proxy.rst]: reverse

File: request/load_balancer_reverse_proxy.rst [hunk 5]
```diff
@@ -7,66 +7,45 @@ an AWS Elastic Load Balancer) or a reverse proxy (e.g. Varnish for
...
-            <framework:config trusted-proxies="192.0.0.1, 10.0.0.0/8">
-                <!-- ... -->
-            </framework:config>
-        </container>
+        // trust *all* "X-Forwarded-*" headers
+        Request::HEADER_X_FORWARDED_ALL
 
-    .. code-block:: php
+        // or, if your proxy instead uses the "Forwarded" header
+        // Request::HEADER_FORWARDED
```
Comment (xabbuh) [inline request/load_balancer_reverse_proxy.rst]: we should also mention the `HEADER_X_FORWARDED_AWS_ELB` constant

**Comment by xabbuh** [comment]
👍 

just some minor comments

File: request/load_balancer_reverse_proxy.rst [hunk 6]
```diff
@@ -7,66 +7,45 @@ an AWS Elastic Load Balancer) or a reverse proxy (e.g. Varnish for
...
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:framework="http://symfony.com/schema/dic/symfony"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                http://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+    // tell Symfony about your revers proxy
+    Request::setTrustedProxies(
+        // the ip address (or range) of your proxy
```
Comment (javiereguiluz) [inline request/load_balancer_reverse_proxy.rst]: `ip address` -> `IP address`

File: request/load_balancer_reverse_proxy.rst [hunk 7]
```diff
@@ -7,66 +7,45 @@ an AWS Elastic Load Balancer) or a reverse proxy (e.g. Varnish for
...
-You are also saying that you trust that the proxy does not send conflicting
-headers, e.g. sending both ``X-Forwarded-For`` and ``Forwarded`` in the same
-request.
-
-That's it! Symfony will now look for the correct headers to get information
-like the client's IP address, host, port and whether the request is
-using HTTPS.
+The Request object has several ``Request::HEADER_*`` constants that control exactly
+*which* headers from your reverse proxy are trusted. The argument is a bit field,
+so you can also pass your own value (e.g. ``0b00110``).
```
Comment (javiereguiluz) [inline request/load_balancer_reverse_proxy.rst]: I'm not sure about this part: `... so you can also pass your own value (e.g. 0b00110)`  Don't you think that'd be super confusing and hard to maintain?

**Review by javiereguiluz — APPROVED** [review]
Ryan, thanks for taking care of this important update. I like it a lot!

**Pull request merged**